### PR TITLE
Source measurement selection API fix. 

### PR DESCRIPTION
**Pull request by v12ganesh**
## Proposed changes

Please include a summary of the changes and any links to related issues. Please also include relevant motivation and context.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Functionality update (non-breaking change which updates or changes existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CI/CD update (an update to the CI/CD workflows, scripts, and/or configurations)
- [ ] Documentation update (an update to enhance the user experience when reading through the docs)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_devices/blob/main/CONTRIBUTING.md) document
- [x] I have signed the CLA
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_devices/pulls) for the same update/change
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Basic linting passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added necessary documentation (if appropriate)
- [ ] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

**Review by u625355 — APPROVED** [review]

File: src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py [hunk 1]
```diff
@@ -44,6 +44,8 @@
     - MEASUrement:MEAS<x>:MAXimum?
     - MEASUrement:MEAS<x>:MEAN?
     - MEASUrement:MEAS<x>:MINImum?
+    - MEASUrement:MEAS<x>:SOUrce1 {CH<x>|MATH|R<x>|D<x>|HIStogram|RF_AMPlitude| RF_FREQuency|RF_PHASe}
```
Comment (nfelt14) [inline src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py:47]: It seems like there is an extra space in this set of enumerations.

File: src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py [hunk 2]
```diff
@@ -2767,6 +2815,51 @@ def minimum(self) -> MeasurementMeasItemMinimum:
...
 
+    @property
+    def source1(self) -> MeasurementMeasItemSource1:
+        """Return the ``MEASUrement:MEAS<x>:SOUrce1`` command.
+
+        **Description:**
+            - For SOURce1: This command specifies the source for all single channel measurements.
+              For delay or phase measurements, This command specifies the waveform to measure
+              'from'. This is equivalent to setting the 'From:' waveform in the 'Measure Delay' side
+              menu or the 'Measure Phase' side menu. SOUrce is equivalent to SOURCE1. For SOUrce2:
```
Comment (nfelt14) [inline src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py:2826]: Why is this referencing SOUrce2 when source 2 isn't available? The syntax appears to be hard-coded to source1.

**Review by nfelt14 — CHANGES_REQUESTED** [review]

**Review by nfelt14 — CHANGES_REQUESTED** [review]
The changelog should be updated

File: src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py [hunk 3]
```diff
@@ -2370,6 +2372,46 @@ class MeasurementMeasItemState(SCPICmdWrite, SCPICmdRead):
...
+    **Info:**
+        - ``CH<x>`` is an analog channel to use as the source waveform. x has a minimum of 1 and a
+          maximum of 4.
+        - ``MATH`` is the math waveform.
+        - ``REF<x>`` is a reference waveform to use as the source waveform. x has a minimum of 1 and
+          a maximum of 4.
+        - ``D<x>`` is a digital channel to use as the source waveform. (MSO/MDO4000/B models only as
+          well as MDO3000 and MDO400C models with option MDO3MSO or MDO4MSO installed.) x has a
+          minimum of 0 and a maximum of 15.
+        - ``HIStogram`` indicates the histogram is the object to be measured. HIStogram only applies
```
Comment (nfelt14) [inline src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py:2407]: This needs references to source2 removed.

File: src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py [hunk 4]
```diff
@@ -2767,6 +2811,47 @@ def minimum(self) -> MeasurementMeasItemMinimum:
...
+        **Info:**
+            - ``CH<x>`` is an analog channel to use as the source waveform. x has a minimum of 1 and
+              a maximum of 4.
+            - ``MATH`` is the math waveform.
+            - ``REF<x>`` is a reference waveform to use as the source waveform. x has a minimum of 1
+              and a maximum of 4.
+            - ``D<x>`` is a digital channel to use as the source waveform. (MSO/MDO4000/B models
+              only as well as MDO3000 and MDO400C models with option MDO3MSO or MDO4MSO installed.)
+              x has a minimum of 0 and a maximum of 15.
+            - ``HIStogram`` indicates the histogram is the object to be measured. HIStogram only
```
Comment (nfelt14) [inline src/tm_devices/commands/_1mlt9u_mdomsodpo/measurement.py:2847]: Remove reference to source2.

**Review by nfelt14 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -22,6 +22,10 @@ Things to be included in the next release go here.
 
 - Updated the `get_model_series()` function to only warn the user if the model is not found in the `SupportedModels` enumeration. This also eliminates false warnings during unit tests.
 
+### Fixed
+
+- fix: Updated the measurement source selection command for the MDO 3K, MDO4K , MSO4K and DPO4K models to work properly.
```
Comment (u625355) [inline CHANGELOG.md:27]: fix: Updated the measurement source selection command for the MDO 3K, MDO4K, MSO4K and DPO4K models to work properly.

Comment (u625355) [inline CHANGELOG.md:27]:
```suggestion
- fix: Updated the measurement source selection command for the MDO3K, MDO4K, MSO4K and DPO4K models to work properly.
```

**Review by u625355 — COMMENTED** [review]
fix: Updated the measurement source selection command for the MDO 3K, MDO4K, MSO4K and DPO4K models to work properly.

**Review by u625355 — APPROVED** [review]

**Review by nfelt14 — APPROVED** [review]

**Comment by nfelt14** [comment]
@v12ganesh, @u625355 please go in and sign all of the commits.

File: CHANGELOG.md [hunk 6]
```diff
@@ -22,6 +22,10 @@ Things to be included in the next release go here.
 
 - Updated the `get_model_series()` function to only warn the user if the model is not found in the `SupportedModels` enumeration. This also eliminates false warnings during unit tests.
 
+### Fixed
+
+- fix: Updated the measurement source selection command for the MDO3K, MDO4K, MSO4K and DPO4K models to work properly.
```
Comment (nfelt14) [inline CHANGELOG.md:27]:
```suggestion
- Updated the measurement source selection command for the MDO3K, MDO4K, MSO4K and DPO4K models to work properly.
```

**Review by nfelt14 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**